freedomofpress/securedrop-sdk
# Add black and isort

Resolves #123

Adds the black and isort tools to the development requirements, and adds Makefile targets for both checking and applying them.

Removes the apparently unused Sphinx requirement, which would otherwise need to be upgraded to address a jQuery security problem.

I'm configuring both black and isort in `pyproject.toml`, in the repo root, because according to [PEP 517](https://www.python.org/dev/peps/pep-0517/#source-trees), "If the pyproject.toml file is absent, or the build-backend key is missing, the source tree is not using this specification, and tools should revert to the legacy behaviour of running setup.py (either directly, or by implicitly invoking the `setuptools.build_meta:__legacy__ backend`)."

The configuration is using [Black's suggested isort settings for compatibility](https://github.com/psf/black/blob/master/docs/compatible_configs.md#isort), as well as an additional [isort setting](https://github.com/timothycrosley/isort/pull/1000) that will appear in an upcoming release.

## Testing

- Check out the first commit on `black-isort` with `git checkout 24dc1ea`
- Run `make check-isort`. It should show the changes it would make.
- Run `make check-black`. It should show the changes it would make.
- Check out the last commit: `git checkout black-isort`
- Rerun `make check-isort` and `make check-black`. No changes should be indicated.
- Run `git blame sdclientapi/__init__.py | head -21 | grep -c a31be70d`. There should be seven lines attributed to that revision.
- Run: 
  - `git config blame.ignoreRevsFile .git-blame-ignore-revs`
  - `git config blame.markIgnoredLines true`
  - `git config blame.markUnblamableLines true`
- Run `git blame sdclientapi/__init__.py | head -21 | grep -c a31be70d`. There should no lines attributed to a31be70d, now that the formatting revision is being ignored.
